examples/platforms/simulation/infra_if.c

```diff
@@ -178,6 +184,21 @@ bool otPlatInfraIfHasAddress(uint32_t aInfraIfIndex, const otIp6Address *aAddres
     return addressesMatch(aAddress, &sIp6Address);
 }
 
+bool otPlatInfraIfHasOnLinkPrefix(uint32_t aInfraIfIndex, const otIp6Address *aAddress)
+{
+    OT_UNUSED_VARIABLE(aInfraIfIndex);
+
+    otIp4Address ip4Addr;
+    bool         isMappedAddress = (otIp4FromIp4MappedIp6Address(aAddress, &ip4Addr) == OT_ERROR_NONE);
+
+    if (isMappedAddress)
+    {
+        return true;
+    }
+
+    return addressesPrefixMatch(aAddress, &sIp6Address);
```

Comment:
How about this? So we avoid `multiple `return` from same function?

```suggestion
    otIp4Address ip4Addr;
    bool         isMappedAddress = (otIp4FromIp4MappedIp6Address(aAddress, &ip4Addr) == OT_ERROR_NONE);

    return isMappedAddress ? true : addressesPrefixMatch(aAddress, &sIp6Address)
```

Reply:
Sure thing, I will refactor this.